microservice/src/main/java/com/skytask/listener/Listener.java

```diff
@@ -0,0 +1,43 @@
+package com.skytask.listener;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.skytask.common.Channels;
+import com.skytask.channel.ProductSource;
+import com.skytask.common.Product;
+import com.skytask.common.ProductMapper;
+import com.skytask.service.ProductService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.messaging.handler.annotation.SendTo;
+
+import java.util.List;
+
+@EnableBinding(ProductSource.class)
+public class Listener {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(Listener.class);
+    private ProductService productService;
+
+    public Listener(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @StreamListener(Channels.CREATE_PRODUCT)
+    @SendTo(Channels.RESPONSE_PRODUCT_LIST)
+    public String createProduct(Product product) throws JsonProcessingException {
+        LOGGER.info("I received: {}", product);
+        productService.create(product);
+        List<Product> products = productService.getProducts();
+        return new ProductMapper().productList2Json(products);
```

Comment:
You're are creating an instance for every messages exchanged. if we have a 1000 messages by minute, would this instantiation have a performance cost?

Reply:
I think it would, I will make it static so I dont have to make an instance.

Reply:
is there any chance of a race condition if you just simply put it as a static field?

Reply:
I plan to use a final static ObjectMapper and use it in ProductMapper. In this scenario I think it wouldn't matter because in ProductMapper I am not changing something critical, i am just using them as a tool. I am not sure if I understood your question correctly.

Reply:
Also I am using ObjectMapper and I read that Mapper instances are fully thread-safe.

Reply:
Nice answer, thx for the research. Go ahead.